### PR TITLE
Entire development of Member Only app

### PR DESCRIPTION
**Pull request by karmaester**
Authentication worked with devise gem
Navbar styled with Bulma
Tables for posts and users with reference by user_id

### CODE REVIEW

File: config/routes.rb
Comment (ghost) [inline config/routes.rb]: Use the `:only` option to define the routes you need to be created for posts resource. According to requirements, you only need the create, new and index actions. See[ screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from Odin.

File: app/controllers/posts_controller.rb
Comment (ghost) [inline app/controllers/posts_controller.rb]: Please remove this action. It is not a requirement for the project. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from Odin. You might wonder why it is important so I will explain. In your routes file, you have to use `:only` option for the specified routes defined in the screenshot. Therefore, the other  ones I am asking you to remove will be redundant.

File: app/controllers/posts_controller.rb
Comment (ghost) [inline app/controllers/posts_controller.rb]: Remove this action.

File: app/controllers/posts_controller.rb
Comment (ghost) [inline app/controllers/posts_controller.rb]: Remove the above two actions since they are not part of the requirements.

File: app/controllers/posts_controller.rb
Comment (ghost) [inline app/controllers/posts_controller.rb]: You should define your own current_user and authenticate_user! helper methods instead of using the ones from devise gem. Define them in the `application_controller.rb`. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png) from Odin.

File: app/views/posts/index.html.erb
Comment (ghost) [inline app/views/posts/index.html.erb]: Avoid using logic in your view files. Any conditional, loops or methods logic should be encapsulated in View helpers `(app/helpers/*_helper.rb)`. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from pathwright.

File: app/views/posts/_form.html.erb
Comment (ghost) [inline app/views/posts/_form.html.erb]: Remove this conditional logic as well.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

### Hello team 😄,

**Your project is well done 💪. Please make the changes highlighted in the comments below before it can be approved.**

## Happy coding 😃.

**PS: You can reach me on slack (Rose Wanjohi)**

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats team, you guys did an excellent job, you have my approval to go ahead and merge this pull request!


## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request closed without merging**
